cadence/contracts/bridge/FlowEVMBridgeUtils.cdc

```diff
@@ -725,6 +747,155 @@ contract FlowEVMBridgeUtils {
         return self.ufix64ToUInt256(value: amount, decimals: self.getTokenDecimals(evmContractAddress: erc20Address))
     }
 
+    /// Gets the declared Cadence contract address declared by an EVM contract in conformance to the ICrossVM.sol
+    /// contract interface. Reverts if the EVM call is unsuccessful.
+    /// NOTE: Just because an EVM contract declares an association does not mean it it is valid!
+    ///
+    /// @param evmContract: The ICrossVM.sol conforming EVM contract from which to retrieve the declared Cadence
+    ///     contract address
+    ///
+    /// @return The resulting Cadence Address as declared associated by the provided EVM contract or nil if the call fails
+    ///
+    access(all)
+    fun getDeclaredCadenceAddressFromCrossVM(evmContract: EVM.EVMAddress): Address? {
+        let cadenceAddrRes = self.call(
+            signature: "getCadenceAddress()",
+            targetEVMAddress: evmContract,
+            args: [],
+            gasLimit: FlowEVMBridgeConfig.gasLimit,
+            value: 0.0
+        )
+        if cadenceAddrRes.status != EVM.Status.successful {
+            return nil
+        }
+        let decodedCadenceAddr = EVM.decodeABI(types: [Type<String>()], data: cadenceAddrRes.data)
+        assert(decodedCadenceAddr.length == 1)
+        var cadenceAddrStr = decodedCadenceAddr[0] as! String
+        if cadenceAddrStr[1] != "x" {
+            cadenceAddrStr = "0x".concat(cadenceAddrStr)
+        }
+        return Address.fromString(cadenceAddrStr) ?? nil
+    }
+
+    /// Gets the declared Cadence Type declared by an EVM contract in conformance to the ICrossVM.sol contract
+    /// interface. Reverts if the EVM call is unsuccessful.
+    /// NOTE: Just because an EVM contract declares an association does not mean it it is valid!
+    ///
+    /// @param evmContract: The ICrossVM.sol conforming EVM contract from which to retrieve the declared Cadence
+    ///     Type
+    ///
+    /// @return The resulting Cadence Type as declared associated by the provided EVM contract or nil if the call fails
+    ///
+    ///
+    access(all)
```

Comment:
I just realized this, but I assume that if we do an EVM call, we can't mark a function as `view` even if it is to a read-only EVM function?

Reply:
Yes, that's right